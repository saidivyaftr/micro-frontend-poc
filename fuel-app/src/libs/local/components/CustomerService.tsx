import colors from '@/shared-ui/colors'
import { makeStyles } from '@material-ui/core'
import TwoColumnLayout from './TwoColumnLayout'

const CustomerService = ({ data }: any) => {
  const classes = useStyles()
  const { leftContent, rightContent }: any = data || {}

  if (!data || Object.keys(data)?.length === 0) {
    return null
  }

  return (
    <div id="customer-service" className={classes.root}>
      <div className={classes.wrapper}>
        <TwoColumnLayout
          leftContent={leftContent?.value}
          rightContent={rightContent?.value}
          leftClass={classes.leftClass}
          rightClass={classes.rightClass}
          wrapperClass={classes.wrapperClass}
        />
      </div>
    </div>
  )
}

const useStyles = makeStyles((theme) => ({
  root: {
    background: colors.main.greenishBlue,
  },
  wrapper: {
    boxSizing: 'content-box',
    maxWidth: 1200,
    margin: '0 auto',
    padding: '3.25rem 1rem',
    [theme.breakpoints.up('md')]: {
      padding: '7.625rem 4rem 5rem',
    },
  },
  leftClass: {
    [theme.breakpoints.up('md')]: {
      flex: 'unset',
      minWidth: '32%',
    },
  },
  rightClass: {
    [theme.breakpoints.up('md')]: {
      flex: 'unset',
      paddingLeft: '40px',
    },
  },
  wrapperClass: {
    alignItems: 'center',
  },
}))

export default CustomerService
