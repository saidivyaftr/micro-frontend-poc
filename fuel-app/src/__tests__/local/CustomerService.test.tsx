import CustomerService from 'src/libs/local/components/CustomerService'
import { render, cleanup } from '@testing-library/react'

const customerServiceData = {
  leftContent: { value: 'left content' },
  rightContent: { value: 'right content' },
}

describe('CustomerService', () => {
  afterEach(cleanup)
  it('should render correctly', () => {
    const { getByText, asFragment } = render(
      <CustomerService data={customerServiceData} />,
    )
    expect(asFragment()).toMatchSnapshot()
    expect(getByText(customerServiceData.leftContent.value)).toBeInTheDocument()
    expect(
      getByText(customerServiceData.rightContent.value),
    ).toBeInTheDocument()
  })

  it('should not render correctly', () => {
    const { queryByText, asFragment } = render(<CustomerService data={''} />)
    expect(asFragment()).toMatchSnapshot()
    expect(
      queryByText(customerServiceData.leftContent.value),
    ).not.toBeInTheDocument()
    expect(
      queryByText(customerServiceData.rightContent.value),
    ).not.toBeInTheDocument()
  })
})
