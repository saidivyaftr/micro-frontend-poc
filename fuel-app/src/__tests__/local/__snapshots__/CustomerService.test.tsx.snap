// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerService should not render correctly 1`] = `<DocumentFragment />`;

exports[`CustomerService should render correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
    id="customer-service"
  >
    <div
      class="makeStyles-wrapper-2"
    >
      <div
        id="two-column-layout"
      >
        <div
          class="makeStyles-wrapper-6 makeStyles-wrapperClass-5"
        >
          <h2
            class="makeStyles-leftContent-7 makeStyles-leftClass-3"
          >
            left content
          </h2>
          <div
            class="makeStyles-rightContent-8 makeStyles-rightClass-4"
          >
            <div
              class="makeStyles-title-9"
            >
              right content
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
